adapters/repos/db/vector/hnsw/restore_test.go

```diff
@@ -0,0 +1,61 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package hnsw
+
+import (
+	"context"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/common"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/distancer"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/testinghelpers"
+	"github.com/weaviate/weaviate/entities/cyclemanager"
+	ent "github.com/weaviate/weaviate/entities/vectorindex/hnsw"
+)
+
+func Test_RestartFromZeroSegments(t *testing.T) {
+	efConstruction := 64
+	ef := 32
+	maxNeighbors := 32
+	dimensions := 20
+	vectors_size := 10000
+	queries_size := 1
+	vectors, _ := testinghelpers.RandomVecs(vectors_size, queries_size, dimensions)
+	distancer := distancer.NewL2SquaredProvider()
+	uc := ent.UserConfig{}
+	uc.MaxConnections = maxNeighbors
+	uc.EFConstruction = efConstruction
+	uc.EF = ef
+	uc.VectorCacheMaxObjects = 10e12
+	uc.PQ = ent.PQConfig{Enabled: true, Encoder: ent.PQEncoder{Type: "kmeans", Distribution: "normal"}}
+	config := Config{
+		RootPath:              path.Join(".", "compression_tests", "fixtures", "restart-from-zero-segments"),
```

Comment:
I would create TempDir, copy test files there and set path to TempDir.
Currently, as those files are modified by inited index, files will be listed as modified by git after running tests locally.